tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,114 @@
+package charts
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	bv1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/rancher/tests/v2/actions/charts"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	shepCharts "github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type BackupTestSuite struct {
+	suite.Suite
+	client   *rancher.Client
+	session  *session.Session
+	S3Client *s3.Client
+}
+
+func (b *BackupTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BackupTestSuite) SetupSuite() {
+	b.session = session.NewSession()
+
+	client, err := rancher.NewClient("", b.session)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+}
+
+func (b *BackupTestSuite) TestS3InPlaceRestore() {
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+
+	project, err := projects.GetProjectByName(b.client, cluster, "System")
+	require.NoError(b.T(), err)
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Info("Checking if the backup chart is already installed...")
+	initialBackupChart, err := shepCharts.GetChartStatus(b.client, project.ClusterID, "cattle-resources-system", "rancher-backup")
+	require.NoError(b.T(), err)
+
+	if !initialBackupChart.IsAlreadyInstalled {
+		installBroChart(b.client)
+	}
```

Comment:
personally think this should be a separate test. You may not have to write a new Test for it though if that makes sense. i.e. you could do a `b.Run(name, func(){...` where the name of the test would be changed before it actually starts installing / provisioning. 

I'm pointing this out because, typically we want to be very specific about what is tested, so combining the use case of 'BRO chart already installed' with 'fresh BRO install' can cause confusion in the future if we don't have that context in the test reporting somehow.

Reply:
That makes sense to me, though I copied this logic from existing chart tests and just made it work for the BRO chart. I can use the `b.run` suggestion though.

Reply:
just checking, were you going to update this or leave it for now?